client/src/containers/transactions/components/form/index.tsx

```diff
@@ -0,0 +1,364 @@
+/* eslint @typescript-eslint/indent: 0,
+   react/jsx-curly-newline: 0 */ // Linter and prettier conflicts, should be fixed.
+import React from 'react';
+import {
+  TextField,
+  Button,
+  Container,
+  Select,
+  MenuItem,
+  InputLabel,
+  FormControl,
+  FormHelperText,
+} from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import DateFnsUtils from '@date-io/date-fns';
+import { State } from 'store';
+import {
+  KeyboardDatePicker,
+  MuiPickersUtilsProvider,
+} from '@material-ui/pickers';
+import {
+  TransactionCategories,
+  Transaction,
+  createTransaction,
+  TransactionFormErrors,
+  setTransactionFormErrors,
+  clearTransactionFormErrors,
+  editTransaction,
+} from 'store/modules/transactions';
+import Routes from 'utils/routes';
+import {
+  AccountSelectOption,
+  getAccountSelectOptions,
+} from 'store/modules/accounts';
+import style from './style.module.scss';
+
+interface IState {
+  amount: number | '';
+  date: Date;
+  category: TransactionCategories | '';
+  description: string;
+  accountOption: AccountSelectOption | null;
+  accountSelectInput: string;
+}
+
+const TransactionForm: React.FC = () => {
+  const [state, setState] = React.useState<IState>({
+    amount: '',
+    date: new Date(),
+    description: '',
+    category: '',
+    accountOption: null,
+    accountSelectInput: '',
+  });
+
+  const history = useHistory();
+  const changeRoute = (route: Routes) => {
+    history.push(route);
+  };
+
+  // handlers are all the same, should be generalized perhaps
+  const handleDateChange = (date: Date) => {
+    setState((prevState) => ({
+      ...prevState,
+      date,
+    }));
+  };
+
+  const handleAmountChange = (amount: number) => {
+    setState((prevState) => ({
+      ...prevState,
+      amount,
+    }));
+  };
+
+  const handleCategoryChange = (category: TransactionCategories) => {
+    setState((prevState) => ({
+      ...prevState,
+      category,
+    }));
+  };
+
+  const handleDescriptionChange = (description: string) => {
+    setState((prevState) => ({
+      ...prevState,
+      description,
+    }));
+  };
```

Comment:
wat if do dis?
`const handleDataChange = (key: string, value: any) => {
    setState((prevState) => ({
        ...prevState,
        [key]: value,
    }));
}`

Reply:
Kaip ir gera idėja, išskyrus any type, nesugalvojau kaip išspręst šitą greitai

Reply:
Na manau vietoj any type nelabai kažkokį kitą ir gali nurodyt šiuo atveju. Am I right or am I right

Reply:
Gali bet tada type declarations užims tiek pat vietos kiek šiti handleriai, manau any negerai naudoti

Reply:
Ye tą ir turėjau omeny, kad nėra trumpai aprašomo varianto, kad apsimokėtų juo pakeisti esamus handlerius. Jei nenori any naudoti, tada skippinam šitą.

Reply:
Jei norime naudoti TS, tai geriau toks kodo dubliavimas nei any times manau, dar geriau turbūt būtų type declaration vienas geras